sycl/source/detail/jit_compiler.cpp

```diff
@@ -820,10 +820,11 @@ jit_compiler::fuseKernels(QueueImplPtr Queue,
 
   ::jit_compiler::BinaryFormat TargetFormat = getTargetFormat(Queue);
   JITConfig.set<::jit_compiler::option::JITTargetFormat>(TargetFormat);
-
+  const std::string tmp = FusedKernelName.str();
+  const char *FusedKernelNametmp = tmp.c_str();
```

Comment:
This sequence seems strange. If we need to create a `std::string` here anyway, there's no point in using char* in this part of the code.

Reply:
That would create an `std::string` temporary anyway. So why not just work with strings here then?